In [57]:
import pandas as pd
import numpy as np

In [62]:
data = pd.read_excel('Sexual Assault Prevention Questionaire (Responses).xlsx')
data.head()
data.shape
(data.columns)
data.index

RangeIndex(start=0, stop=1004, step=1)

In [9]:
data.head()

Timestamp What is your current age?  \
0 2019-04-17 13:38:40.860                     35-45   
1 2019-04-17 19:17:05.661                     18-25   
2 2019-04-17 19:54:41.168                     18-25   
3 2019-04-17 16:39:57.570                     18-25   
4 2019-04-17 19:06:38.610                     18-25   

  Which of the categories best describe you? How do you identify yourself?  \
0                                      White                          Male   
1                     Asian/Pacific Islander                        Female   
2                     Asian/Pacific Islander                        Female   
3                                      Black                        Female   
4                                      Other                          Male   

  Do you own or carry any forms of protection in case of sexual assault, attack, or other potential harm?  \
0          No, I do not own any forms of protection.                                                        
1  No, I do not own or carry any forms of protect...                                                        
2  No, I do not own or carry any forms of protect...                                                        
3  No, I do not own or carry any forms of protect...                                                        
4  No, I do not own or carry any forms of protect...                                                        

  Would you buy a product that would help protect you before sexual assault was to occur?  \
0  No, I would not buy something to protect me ag...                                        
1  Maybe, I would think about buying a product to...                                        
2  Maybe, I would think about buying a product to...                                        
3  Maybe, I would think about buying a product to...                                        
4  No, I would not buy something to protect me ag...                                        

  Have you ever used any commercial DNA testing services (23andMe, Ancestry, etc)?  \
0                                                 No                                 
1                                                 No                                 
2                                                 No                                 
3                                                 No                                 
4                                                 No                                 

  Have you ever been a victim of sexual assault or rape?  \
0                                                 No       
1                                                 No       
2                                                Yes       
3                                                 No       
4                                                 No       

  Have you ever witnessed assault or rape?  \
0                                      NaN   
1                                       No   
2                                       No   
3                                       No   
4                                       No   

  If you answered "Yes" or "Maybe" to "Have you ever been a victim of sexual assault or rape?" or "Have you ever witnessed assault or rape?", did you or the victim file a police report or go to the hospital afterwards?  \
0                                                NaN                                                                                                                                                                         
1                                                NaN                                                                                                                                                                         
2  No, I  nor the victim did not go to the hospit...                                                                                                                                 

In [64]:
renam= {'What is your current age?':'Age', 'Which of the categories best describe you?':'Category',
       'How do you identify yourself?':'Identification'}
dataa = data.rename(columns=renam)
data.head()

Timestamp What is your current age?  \
0 2019-04-17 13:38:40.860                     35-45   
1 2019-04-17 19:17:05.661                     18-25   
2 2019-04-17 19:54:41.168                     18-25   
3 2019-04-17 16:39:57.570                     18-25   
4 2019-04-17 19:06:38.610                     18-25   

  Which of the categories best describe you? How do you identify yourself?  \
0                                      White                          Male   
1                     Asian/Pacific Islander                        Female   
2                     Asian/Pacific Islander                        Female   
3                                      Black                        Female   
4                                      Other                          Male   

  Do you own or carry any forms of protection in case of sexual assault, attack, or other potential harm?  \
0          No, I do not own any forms of protection.                                                        
1  No, I do not own or carry any forms of protect...                                                        
2  No, I do not own or carry any forms of protect...                                                        
3  No, I do not own or carry any forms of protect...                                                        
4  No, I do not own or carry any forms of protect...                                                        

  Would you buy a product that would help protect you before sexual assault was to occur?  \
0  No, I would not buy something to protect me ag...                                        
1  Maybe, I would think about buying a product to...                                        
2  Maybe, I would think about buying a product to...                                        
3  Maybe, I would think about buying a product to...                                        
4  No, I would not buy something to protect me ag...                                        

  Have you ever used any commercial DNA testing services (23andMe, Ancestry, etc)?  \
0                                                 No                                 
1                                                 No                                 
2                                                 No                                 
3                                                 No                                 
4                                                 No                                 

  Have you ever been a victim of sexual assault or rape?  \
0                                                 No       
1                                                 No       
2                                                Yes       
3                                                 No       
4                                                 No       

  Have you ever witnessed assault or rape?  \
0                                      NaN   
1                                       No   
2                                       No   
3                                       No   
4                                       No   

  If you answered "Yes" or "Maybe" to "Have you ever been a victim of sexual assault or rape?" or "Have you ever witnessed assault or rape?", did you or the victim file a police report or go to the hospital afterwards?  \
0                                                NaN                                                                                                                                                                         
1                                                NaN                                                                                                                                                                         
2  No, I  nor the victim did not go to the hospit...                                                                                                                                 

In [43]:
data.dtypes

Timestamp                                                                                                                                                                                                                   datetime64[ns]
What is your current age?                                                                                                                                                                                                           object
Which of the categories best describe you?                                                                                                                                                                                          object
How do you identify yourself?                                                                                                                                                                                                       object
Do you own or carry any forms of protection in case of sexua

In [41]:
data.get_dtype_counts()

float64            1
datetime64[ns]     1
object            10
dtype: int64

In [12]:
data.isnull().sum()

Timestamp                                                                                                                                                                                                                     0
What is your current age?                                                                                                                                                                                                     0
Which of the categories best describe you?                                                                                                                                                                                    0
How do you identify yourself?                                                                                                                                                                                                 0
Do you own or carry any forms of protection in case of sexual assault, attack, or other potential harm? 

In [31]:
data.iloc[:, 3].value_counts()

Female               537
Male                 461
Prefer not to say      3
agender                1
Nonbinary              1
Genderqueer            1
Name: How do you identify yourself?, dtype: int64

*There are 537 females, 461 males, 3 people did'nt specify, 1 agender, 1 nonbinary and 1 genderqueer*


In [50]:
data.iloc[:, 11].groupby(data.iloc[:, 3])+

In [59]:
data.describe(include=[np.number]).T

count      mean  \
How comfortable do you feel going to the author...  1003.0  6.481555   

                                                         std  min  25%  50%  \
How comfortable do you feel going to the author...  2.812931  1.0  5.0  7.0   

                                                    75%   max  
How comfortable do you feel going to the author...  9.0  10.0

In [61]:
data.describe(include=np.object).T

count unique  \
What is your current age?                           1004      6   
Which of the categories best describe you?          1004     17   
How do you identify yourself?                       1004      6   
Do you own or carry any forms of protection in ...  1004     20   
Would you buy a product that would help protect...  1004      4   
Have you ever used any commercial DNA testing s...  1004      2   
Have you ever been a victim of sexual assault o...  1004      3   
Have you ever witnessed assault or rape?            1003      2   
If you answered "Yes" or "Maybe" to "Have you e...   444      4   
If you did not file a police report or go to th...   230    211   

                                                                                                  top  \
What is your current age?                                                                       25-35   
Which of the categories best describe you?                                                      White   
How do you identify yourself?                                                                  Female   
Do you own or carry any forms of protection in ...  No, I do not own or carry any forms of protect...   
Would you buy a product that would help protect...  Yes, I would buy a product that would help pro...   
Have you ever used any commercial DNA testing s...                                                 No   
Have you ever been a victim of sexual assault o...                                                 No   
Have you ever witnessed assault or rape?                                                           No   
If you answered "Yes" or "Maybe" to "Have you e...  No, I  nor the victim did not go to the hospit...   
If you did not file a police report or go to th...                                             Scared   

                                                   freq  
What is your current age?                           442  
Which of the categories best describe you?          547  
How do you identify yourself?                       537  
Do you own or carry any forms of protection in ...  541  
Would you buy a product that would help protect...  465  
Have you ever used any commercial DNA testing s...  779  
Have you ever been a victim of sexual assault o...  677  
Have you ever witnessed assault or rape?            830  
If you answered "Yes" or "Maybe" to "Have you e...  264  
If you did not file a police report or go to th...    3

In [ ]:
data.groupby(data.iloc[:, 3])[]

In [1]:
file = open('example.txt', 'w')
l = ['line1', 'line 2', 'line 3']
for item in l:
    file.write(item + '\n')
file.close()

In [5]:
with open('example.txt', 'w') as file:
    l = ['line1', 'line 2', 'line 3']
    for item in l:
        file.write(item)
